Update rules for inclusion in client list (#1833)
jellyfin/jellyfin.org: blog/2026/05-03-client-inclusion-rules-update/index.mdx

Jellyfin is a project that exists precisely because of its Free Software ideals, and it would be at odds with the ethos of the project to include clients that do not share these ideals.

Note that this does not mean that a developer is prohibited from charging money for a client, though we will evaluate commercial restrictions closely as one component of the following client inclusion rules. But all clients must have their source code available under one of the aforementioned Free Licenses to qualify.

This includes existing clients on the list; namely Infuse, Yatse, Manet, Symfonium, and Finer. All 5 of these will be removed from our list going forward. While Infuse and Yatse, especially, have been listed clients for an extremely long-time, we do not believe that making continued exceptions is worthwhile. In light of our desire to institute these new rules, we want to avoid "loopholes" for evaluating future clients, and the project has simply grown past the point where we feel the need to include every possible client for compatibility reasons, as we did during the early days when the aforementioned clients were added.

> All of this is not to say that we are opposed to the creation of proprietary, closed source clients. You as a client developer are free to choose whatever license you wish for your own project. And you are still **free to post about such clients in our community spaces** (including [`/r/jellyfin`](https://old.reddit.com/r/jellyfin) and the like). We do not want to discourage innovation in the Jellyfin ecosystem. This rule **only concerns inclusion in the official clients list on our website**, which can be seen by new users as our official endorsement. We simply do not wish to explicitly endorse proprietary clients going forward.

Reviewer: ```suggestion
All of this is not to say that we are opposed to the creation of proprietary, closed source clients. You as a client developer are free to choose whatever license you wish for your own project. And you are still **free to post about such clients in our community spaces** (including [`/r/jellyfin`](https://www.reddit.com/r/jellyfin/) and the like). We do not want to discourage innovation in the Jellyfin ecosystem. This rule **only concerns inclusion in the official clients list on our website**, which can be seen by new users as our official endorsement. We simply do not wish to explicitly endorse proprietary clients going forward.
```

Don't link to old reddit

The purpose of this rule is to reduce attempts at including brand-new, unestablished clients until they can prove themselves as being relevant and persistent to the wider community. There are two parts to this rule.

The first is active development time. We chose 1 year as a good ballpark, because this is, to us, about the amount of time it takes for a client to truly mature and for a developer to prove that they are committed to maintaining the client in the long-term. We do not want to list clients that are just throwing something into the ecosystem half-baked and then expecting others to run with it.

The second is user interest. This both complements the 1-year development time requirement, and also offers an exception to it. 

We understand that part of the desire of "being listed" is to drive users towards your client, but you must first put in the work to build a community of users yourself. We don't want the client list to be an advertisement; we want it to provide a list of clients that we believe best exemplify the Jellyfin ecosystem and provide a stable, consistent, and feature-rich experience for users.

> Massive user interest can be an exception to the time requirement. If you are able to demonstrate to the team proof of a sizeable userbase and active development from contributors other than just the original author, we **may** consider waving the 1-year requirement. By our definition, "massive user interest" means at least several thousand active users and significant, persistent, and organic discussion of and contribution to the client within the wider community. Unless you're absolutely certain that your client meets this high threshold, it almost certainly does not. These exceptions will be handled on a case-by-case basis and explicit proof will be required; if you believe you qualify, please [contact the Core team](https://jellyfin.org/contact/).

Reviewer: Note: Core Team Contact is not yet mentioned on the contact page.
I will soon change this though. https://github.com/jellyfin/jellyfin.org/issues/951

The purpose of this rule is to reduce attempts at including brand-new, unestablished clients until they can prove themselves as being relevant and persistent to the wider community. There are two parts to this rule.

The first is active development time. We chose 1 year as a good ballpark, because this is, to us, about the amount of time it takes for a client to truly mature and for a developer to prove that they are committed to maintaining the client in the long-term. We do not want to list clients that are just throwing something into the ecosystem half-baked and then expecting others to run with it.

The second is user interest. This both complements the 1-year development time requirement, and also offers an exception to it. 

We understand that part of the desire of "being listed" is to drive users towards your client, but you must first put in the work to build a community of users yourself. We don't want the client list to be an advertisement; we want it to provide a list of clients that we believe best exemplify the Jellyfin ecosystem and provide a stable, consistent, and feature-rich experience for users.

> Massive user interest can be an exception to the time requirement. If you are able to demonstrate to the team proof of a sizeable userbase and active development from contributors other than just the original author, we **may** consider waving the 1-year requirement. By our definition, "massive user interest" means at least several thousand active users and significant, persistent, and organic discussion of and contribution to the client within the wider community. Unless you're absolutely certain that your client meets this high threshold, it almost certainly does not. These exceptions will be handled on a case-by-case basis and explicit proof will be required; if you believe you qualify, please [contact the Core team](https://jellyfin.org/contact/).

Reviewer 2: Contact link should not have a trailing slash, and I believe a relative URL would be preferred here.

```suggestion
Massive user interest can be an exception to the time requirement. If you are able to demonstrate to the team proof of a sizeable userbase and active development from contributors other than just the original author, we **may** consider waving the 1-year requirement. By our definition, "massive user interest" means at least several thousand active users and significant, persistent, and organic discussion of and contribution to the client within the wider community. Unless you're absolutely certain that your client meets this high threshold, it almost certainly does not. These exceptions will be handled on a case-by-case basis and explicit proof will be required; if you believe you qualify, please [contact the Core team](/contact).
```

The last line is a guard against a client proclaiming that "I'm working on this killer feature", but not actually having it ready. Inclusion can wait until it's ready and actually released to users.

## How to Follow These Requirements

> When creating a pull request for addition to the clients list, we request that you provide detail on how your client meets all of the requirements. This includes things like a link to the GitHub page of the client (with a clear LICENSE definition and Git history), and a list of your interesting features and functionality. Please do not force us to go digging, and do not use an LLM to write this for you - use your own words (as per our [LLM policies](https://jellyfin.org/docs/general/contributing/llm-policies)) and make an active effort to _sell us_ on your client.

Reviewer 2: ```suggestion
When creating a pull request for addition to the clients list, we request that you provide detail on how your client meets all of the requirements. This includes things like a link to the code repository of the client (with a clear LICENSE definition and Git history), and a list of your interesting features and functionality. Please do not force us to go digging, and do not use an LLM to write this for you - use your own words (as per our [LLM policies](https://jellyfin.org/docs/general/contributing/llm-policies)) and make an active effort to _sell us_ on your client.
```

Nowadays it's fairly common to see FOSS projects on alternative platforms, we don't want to give the impression a client must be on github
